api.go

```diff
@@ -558,15 +558,24 @@ func (c Client) newRequest(method string, metadata requestMetadata) (req *http.R
 		method = "POST"
 	}
 
-	var location string
-	// Gather location only if bucketName is present.
-	if metadata.bucketName != "" && metadata.bucketLocation == "" {
+	location := metadata.bucketLocation
+	if metadata.bucketName != "" && location == "" {
+		// Gather location only if bucketName is present.
 		location, err = c.getBucketLocation(metadata.bucketName)
 		if err != nil {
 			return nil, err
 		}
-	} else {
-		location = metadata.bucketLocation
+	}
+	if location == "" {
+		// Default to location to 'us-east-1'.
+		location = "us-east-1"
+		if s3utils.IsAmazonChinaEndpoint(c.endpointURL) {
+			location = "cn-north-1"
+		} else if s3utils.IsAmazonGovCloudEndpoint(c.endpointURL) {
+			location = "us-gov-west-1"
+		} else if c.region != "" {
+			location = c.region
+		}
 	}
```

Comment:
I tried that.. but it got complex. So an if-condition was easier. In-fact will cleanup to put this under functions for each cases.